lms/static/scripts/frontend_apps/components/GradingComment.tsx

```diff
@@ -0,0 +1,156 @@
+import type { TextareaProps } from '@hypothesis/frontend-shared';
+import {
+  Button,
+  CancelIcon,
+  IconButton,
+  NoteFilledIcon,
+  NoteIcon,
+  PointerUpIcon,
+  Textarea,
+  useClickAway,
+  useKeyPress,
+} from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { MutableRef } from 'preact/hooks';
+import {
+  useCallback,
+  useId,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'preact/hooks';
+
+type CommentPopoverProps = {
+  disabled: boolean;
+  comment: string;
+  onInput?: TextareaProps['onInput'];
+  onSubmit?: (e: Event) => Promise<boolean>;
+  closePopover: () => void;
+};
+
+function CommentPopover({
+  disabled,
+  comment,
+  onInput,
+  onSubmit,
+  closePopover,
+}: CommentPopoverProps) {
+  const commentRef = useRef<HTMLTextAreaElement | null>(null);
+  const commentId = useId();
+
+  // Focus comment textarea when popover is open
+  useLayoutEffect(() => {
+    commentRef.current!.focus();
+  }, []);
```

Comment:
Since this is now a component of its own, we can focus it always when rendered, without having to handle state or props.